src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -750,20 +750,24 @@ private static object SplitWithPattern(ExecutionContext context, IScriptExtent e
                 separatorPattern = Regex.Escape(separatorPattern);
             }
 
-            if (limit < 0)
+            RegexOptions regexOptions = parseRegexOptions(options);
+
+            int calculatedLimit = limit;
+
+            // if the limit is negative then set Regex to read from right to left
+            if (calculatedLimit < 0) 
             {
-                // Regex only allows 0 to signify "no limit", whereas
-                // we allow any integer <= 0.
-                limit = 0;
+                regexOptions |= RegexOptions.RightToLeft;
+                calculatedLimit *= -1;
             }
 
-            RegexOptions regexOptions = parseRegexOptions(options);
             Regex regex = NewRegex(separatorPattern, regexOptions);
 
             List<string> results = new List<string>();
```

Comment:
Good opportunity to be DRY
```suggestion
            var results = new List<string>();
```